packages/edit-site/src/components/global-styles/use-global-styles-output.js

```diff
@@ -761,6 +767,22 @@ export const getBlockSelectors = ( blockTypes ) => {
 	return result;
 };
 
+/**
+ * Determines whether or not the theme has disabled all layout styles output.
+ *
+ * This feature only disables the output of layout styles,
+ * the controls for adjusting layout will still be available in the editor.
+ * Themes that use this feature commit to providing their own styling for layout features.
+ *
+ * @return {boolean} Whether or not the theme opts-in to disable all layout styles.
+ */
+function useThemeHasDisabledLayoutStyles() {
```

Comment:
Could the site editor package access the hook from the block editor package instead of duplicating it?

Reply:
That's a good question — I wasn't sure if we wanted to expose it for use anywhere else so left it as a duplication for now. But if it's an official theme support, then yes, it probably makes better sense to have it defined once in the block editor package, and then export it 👍

I'll have a play.

Reply:
After taking another look, I think it makes the code a little clearer if we don't actually use the hook abstraction. It didn't add much, and meant that if you wanted to see what it was actually checking you'd have to look at the implementation of the hook. In https://github.com/WordPress/gutenberg/pull/42544/commits/6b6dadb00dc25571dc0c9474da26e719fe1e1bfb I've switched to inlining the settings checks, so that it's closer to the feature name, which I think on balance, is a little easier to read.

Happy to change it, though, if you think there's a better way to handle it!